File: project01/rectangle_area.rb
# computes the area of rectangle 
#
# Haim Hong
# Jan 15 2022
###############################################################

# function computes area of rectangle, given its width and legnth
# param width legnth
# returns area of rectangle
def rectangleArea(width, length)
    width * length
end

if __FILE__ == $0
   puts "To compute the area of a circle," # prints out string with new line
   print " enter its width: " # prints out string with no new line 
   width = gets.chomp.to_f # read user input wiht chomp removing new line and changes to float
   print " enter its length: " # prints out string with no new line 
   length = gets.chomp.to_f # read user input wiht chomp removing new line and changes to float
   print "The rectangle's area is: " # print out string
   puts rectangleArea(width, length) # prints out result from calling function with new line 
end

File: project01/clojure/src/rectangle_area.clj
;;; it calculates the area of rectangle
;;;
;;; get swidth and legnth of rectangle
;;; outputs the area of rectangle
;;;
;;; Haim Hong
;;; Jan 14 2022
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns rectangle_area)

(defn rectangleArea [itsWidth itsLength]
  (* itsWidth itsLength)
)

(defn -main []
  (println "\nTo compute the area of a rectangle,") ;;; prints string 
  (print   " enter its width: ") (flush)

  (let
   [width (read)] ;;; gets the input of width
    (assert (>= width 0) "-main: width must be positive") ;;; check if the width is not negative
    (print   " enter its length: ") (flush)
    (let
     [length (read)] ;;; gets the input of legnth
      (assert (>= length 0) "-main: length must be positive") ;;; check if the length is not negative
      (printf "\nThe area is %f\n\n" (rectangleArea width length)) ;; prints out the result with new line 

      (print "\nThe area is ") ;;; prints string with new line
      (print (rectangleArea width length)) ;;; prints out the radius using the function
      (print "\n\n") ;;; prints new line

      (printf "\nThe area is %.15f\n\n" (rectangleArea width length)) ;;; to 15th decimal places
      ))
  )